### PR TITLE
Remove LabelSet

### PR DESCRIPTION
**Pull request by c24t**
Following https://github.com/open-telemetry/oteps/pull/90.

@jmacd please check my work here, I may have mangled some of your go examples.

### CODE REVIEW

File: specification/metrics/api-user.md [hunk 1]
```diff
@@ -182,30 +178,33 @@ func newServer(meter metric.Meter) *server {
 func (s *server) operate(ctx context.Context) {
      // ... other work
 
-     s.instruments.counter1.Add(ctx, 1, s.meter.Labels(
-         label1.String("..."),
-         label2.String("...")))
+     s.instruments.counter1.Add(ctx, 1, []KeyValue{
```
Comment (jmacd) [inline specification/metrics/api-user.md:181]: There is a `...` in this call, so you can remove the `[]KeyValue{}` wrapping and just list two arguments.

File: specification/metrics/api-user.md [hunk 2]
```diff
@@ -217,30 +216,28 @@ various trade-offs in terms of complexity and performance.
...
 func (s *server) processStream(ctx context.Context) {
 
-  streamLabels := s.meter.Labels(
-      labelA.String("..."),
-      labelB.String("..."),
-  )
   // The result of Bind() is a bound instrument
   // (e.g., a BoundInt64Counter).
-  counter2 := s.instruments.counter2.Bind(streamLabels)
+  counter2 := s.instruments.counter2.Bind([]KeyValue{
```
Comment (jmacd) [inline specification/metrics/api-user.md:237]: nits: same as above ^^^, and two more below.

File: specification/metrics/api-user.md [hunk 3]
```diff
@@ -85,13 +83,11 @@ include the instrument, a numerical value, and an optional
...
-refers to the key component while "label value" refers to the
-correlated value component of a label.  Label refers to the pair of
-label key and value.  Labels are passed in to the metric event in the
-form of a `LabelSet` argument, using several input methods discussed
-below.
+Labels are key:value pairs associated with events describing various dimensions
+or categories that describe the event.  A "label key" refers to the key
+component while "label value" refers to the correlated value component of a
+label.  Label refers to the pair of label key and value.  Labels are passed in
+to the metric event at construction time.
```
[the remark this answers is not in the document]
Comment (jmacd) [reply specification/metrics/api-user.md:90]: (Sorry, as a reminder, this document wasn't updated for 0.3, so there are lots of inconsistencies...)

**Comment by c24t** [comment]
Thanks for the quick reviews, hopefully this change is still worth making even if the whole file will be overhauled soon.

File: specification/metrics/api-user.md [hunk 4]
```diff
@@ -481,25 +395,23 @@ metric events.
 
 ### Interaction with distributed correlation context
 
-The `LabelSet` type introduced above applies strictly to "local"
-labels, meaning provided in a call to `meter.Labels(...)`.  The
-application explicitly declares these labels, whereas distributed
-correlation context labels are implicitly associated with the event.
+As described above, labels are strictly to "local".  I.e., the application
```
Comment (arminru) [inline specification/metrics/api-user.md:398]:
```suggestion
As described above, labels are strictly "local".  I.e., the application
```

**Comment by jmacd** [comment]
Re: line-breaks 
Many of these are my doing. I didn't realize there was a soft-breaks policy. @c24t should we remove hard line-breaks in a separate PR across the metrics specs (or the whole repo)?

**Comment by c24t** [comment]
Sorry about the line breaks @arminru, I re-wrapped paragraphs I edited which made for a lot of spurious line diffs.

> should we remove hard line-breaks in a separate PR across the metrics specs (or the whole repo)?

I'd be happy to, I'll put up a PR.

**Comment by jmacd** [comment]
May we merge this ahead of adding soft-breaks everywhere? 
I'm feeling like drafting spec changes to help push https://github.com/open-telemetry/oteps/pull/93 and https://github.com/open-telemetry/oteps/pull/96 forward, but I'd like this merged first.

**Pull request merged**